### PR TITLE
feat: [PL-23039]: Support for GitSync with Bitbucket

### PR DESCRIPTION
**Pull request by sunnykesh-harness**
##### Summary:
Enabling the support for Bitbucket for gitSync

##### Jira Links:
https://harness.atlassian.net/browse/PL-23039

##### Screenshots:
Screenshot 2022-02-24 at 11 31 36 AM

You can use the following comments to re-trigger PR Checks

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress: `trigger cypress`
- Fix Prettier - `fix prettier`

### CODE REVIEW

**Comment by sunnykesh-harness** [comment]
retrigger jest

**Review by mayankVermaHarness — APPROVED** [review]

**Comment by meenakshi-harness** [comment]
Are we not using Thumbnail component here?

**Comment by sunnykesh-harness** [comment]
> 

No we are using Card.

File: src/modules/40-gitsync/components/gitSyncRepoForm/GitSyncRepoForm.tsx [hunk 1]
```diff
@@ -107,6 +110,8 @@ const GitSyncRepoForm: React.FC<ModalConfigureProps & GitSyncRepoFormProps> = pr
     queryParams: { accountIdentifier: accountId }
   })
 
+  const supportedProviders = bitBucketSupported ? gitCards : gitCards.filter(card => card.type !== Connectors.BITBUCKET)
```
Comment (avinashmadhwani02) [inline src/modules/40-gitsync/components/gitSyncRepoForm/GitSyncRepoForm.tsx:113]: this is not scalable, imagine adding one more provider with feature flag check - you will have to nest conditions. 

can be done this way - 

```
const isSupported = (card) => {
 if(card.type === Connectors.BITBUCKET) {
  return Boolean(featureFlagSupported)
 }
  return true; 
}

const supportedProviders = gitCards.filter(card => isSupported(card));

```

or you can even use a switch case/ object.

**Review by avinashmadhwani02 — APPROVED** [review]

File: src/modules/40-gitsync/components/gitSyncRepoForm/GitSyncRepoForm.tsx (hunk 1 above)
Comment (sunnykesh-harness) [reply src/modules/40-gitsync/components/gitSyncRepoForm/GitSyncRepoForm.tsx:113]: Making this change in follow up PR.

**Pull request merged**